tests/test_db.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright 2019 Tyler Neely
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate rocksdb;
+extern crate libc;
+
+mod util;
+
+use libc::{size_t};
+
+use rocksdb::{DB, DBVector, Error, IteratorMode, Options, WriteBatch};
+use std::str;
+use util::DBPath;
+
+#[test]
+fn test_db_vector() {
+    use std::mem;
+    let len: size_t = 4;
+    let data: *mut u8 = unsafe { mem::transmute(libc::calloc(len, mem::size_of::<u8>())) };
+    let v = unsafe { DBVector::from_c(data, len) };
+    let ctrl = [0u8, 0, 0, 0];
+    assert_eq!(&*v, &ctrl[..]);
+}
+
+#[test]
+fn external() {
+  let path = DBPath::new("_rust_rocksdb_externaltest");
+  let db = DB::open_default(&path).unwrap();
+  let p = db.put(b"k1", b"v1111");
+  assert!(p.is_ok());
+  let r: Result<Option<DBVector>, Error> = db.get(b"k1");
+  assert!(r.unwrap().unwrap().to_utf8().unwrap() == "v1111");
+  assert!(db.delete(b"k1").is_ok());
+  assert!(db.get(b"k1").unwrap().is_none());
+}
```

Comment:
I suppose there is something wrong with formatting ?

Reply:
This was just code that was moved from `db.rs` into this file so `DBPath` could be used.  If the formatting is wrong, I suppose it was prior to me moving it around.  That said, I'll try to fix it up...

Reply:
Fixed.